File: vehical/public/js/employee.js
frappe.ui.form.on('Employee', {
    refresh(frm){
        frm.add_custom_button(__("Check In"), () =>{
            frappe.call({
                method:"vehical.public.employee.open_check",
                args: {
                    emp:frm.doc.employee,
                    name:frm.doc.employee_name,
                }
            })
        })
    },
    validate(frm) {
        frappe.confirm("ok", () => {
            frm.save();
        })
    }
});

File: vehical/public/employee.py
import frappe
from frappe.utils import now


@frappe.whitelist()
def open_check(emp,name):
    doc=frappe.new_doc("Employee Checkin")
    doc.employee=emp
    doc.employee_name=name
    doc.log_type="IN"
    doc.save()
    
def validate_emp(doc, action):
    if doc.notice_number_of_days > 90:
        frappe.msgprint("Notice period cannot be greater than 90 days", alert=True, indicator="red")
